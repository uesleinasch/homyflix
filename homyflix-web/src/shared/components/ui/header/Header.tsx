import { Box, Divider, Flex, Title } from "@mantine/core";

interface HeaderProps {
  title: string;

  children?: React.ReactNode;
}

const Header: React.FC<HeaderProps> = ({ title, children }) => {
  return (
    <>
      <Box pl={{base: "0px", sm: "md"}} pr={{base: "0px", sm: "md"}}>
        <Flex justify="space-between" wrap="wrap" align="center" p="0px" gap={{base: "xs", sm: "md"}}>
          <Box>
            <Title order={1} size="h2">
              {title}
            </Title>
          </Box>
          <Box>{children}</Box>
        </Flex>
        <Divider my="md" />
      </Box>
    </>
  );
};

export default Header;
