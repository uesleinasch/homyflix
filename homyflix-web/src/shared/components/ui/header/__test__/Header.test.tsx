import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Header from '../Header';
import { MantineProvider } from '@mantine/core';

describe('Header', () => {
  it('Should render correctly', () => {
    const { container } = render(
        <MantineProvider>
            <Header title="Test Header" />
        </MantineProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('Should render children content correctly', () => {
    const { container } = render(
        <MantineProvider>
            <Header title="Test Header">
                <div>Test content for Header</div>
            </Header>
        </MantineProvider>
    );
    expect(container).toHaveTextContent('Test content for Header');
  });
});