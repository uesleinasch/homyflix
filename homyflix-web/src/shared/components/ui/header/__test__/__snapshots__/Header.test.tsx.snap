// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render correctly 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <style
    data-mantine-styles="inline"
  >
    .__m__-«r0»{padding-left:0rem;padding-right:0rem;}@media(min-width: 48em){.__m__-«r0»{padding-left:var(--mantine-spacing-md);padding-right:var(--mantine-spacing-md);}}
  </style>
  <div
    class="__m__-«r0»"
  >
    <style
      data-mantine-styles="inline"
    >
      .__m__-«r1»{gap:var(--mantine-spacing-xs);}@media(min-width: 48em){.__m__-«r1»{gap:var(--mantine-spacing-md);}}
    </style>
    <div
      class="m_8bffd616 mantine-Flex-root __m__-«r1»"
      style="align-items: center; justify-content: space-between; flex-wrap: wrap; padding: 0rem;"
    >
      <div
        class=""
      >
        <h1
          class="m_8a5d1357 mantine-Title-root"
          data-order="1"
          data-size="h2"
          style="--title-fw: var(--mantine-h2-font-weight); --title-lh: var(--mantine-h2-line-height); --title-fz: var(--mantine-h2-font-size);"
        >
          Test Header
        </h1>
      </div>
      <div
        class=""
      />
    </div>
    <div
      class="m_3eebeb36 mantine-Divider-root"
      data-orientation="horizontal"
      role="separator"
      style="margin-block: var(--mantine-spacing-md);"
    />
  </div>
</div>
`;
